src/ocean/util/encode/Base64.d

```diff
@@ -446,12 +446,17 @@ body
         if (paddedPos)
         {
             auto padded = data[($ - paddedPos) .. $];
-            foreach(char piece; padded)
+            assert (padded.length <= 4);
+
+            // Fill the missing paddings if needed
+            base64Quad[padded.length .. $] = BASE64_PAD;
+
+            foreach(i, char piece; padded)
             {
                 ubyte next = table[piece];
                 if (next || piece == 'A')
                     *quadPtr++ = next;
-                if (quadPtr is endPtr)
+                if (i == padded.length - 1)
```

Comment:
No, that would require that `paddedPos` is zero, so it would not enter this branch. Additionally, the `foreach` wouldn't iterate over `padded`.